### PR TITLE
MOD-2830- Add properties, events and method for TagTechnologyMifareUltralight

### PR DESCRIPTION
**Pull request by anil-shukla-axway**
https://jira.appcelerator.org/browse/MOD-2830

### CODE REVIEW

File: ios/Classes/TiNfcNativeTagTechnologyProxy.m [hunk 1]
```diff
@@ -26,24 +26,31 @@ - (id)_initWithPageContext:(id<TiEvaluator>)context andSession:(NFCTagReaderSess
 
 - (void)connect:(id)unused
 {
-  [self.session connectToTag:self.tag
+  [self.session connectToTag:[self.tag tag]
            completionHandler:^(NSError *_Nullable error) {
              if (![self _hasListeners:@"didConnectTag"]) {
                return;
              }
+             if (error != nil) {
```
Comment (fasihul-axway) [inline ios/Classes/TiNfcNativeTagTechnologyProxy.m:34]: May be we can do like 

```suggestion
             [self fireEvent:@"didConnectTag"
                    withObject:@{
                        @"errorCode" : error != nil ? NUMINTEGER([error code]) : NULL,
                        @"errorDescription" : error != nil ? [error localizedDescription] : NULL,
                        @"errorDomain" : error != nil ? [error domain] : NULL,
                      @"tag" : self.tag
                    }];
```

**Review by fasihul-axway — CHANGES_REQUESTED** [review]

File: apidoc/NfcMiFareTagTechnology.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  - name: historicalBytes
+    summary: The historical bytes extracted from an Answer To Select response.
+
+  - name: sendMiFareCommand
+    summary: Sends a native MIFARE command to the tag.
+    parameters:
+    - name: data
+      summary: hex command with length will be sent in form of data
+      type: Titanium.Buffer
```
Comment (vikas-goyal-axway) [inline apidoc/NfcMiFareTagTechnology.yml:27]: It should be ```Array<Number>```

File: apidoc/NfcMiFareTagTechnology.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      type: String
+      summary: The error domain; only present if an error occurred.
+    - name: errorDescription
+      type: String
+      summary: The error description; only peresent if an error occurred.
+
+  - name: sendMiFareISO7816Command
+    summary: Sends a native MIFARE command to the tag along with other parameter.
+    parameters:
+    - name: sw1
```
Comment (vikas-goyal-axway) [inline apidoc/NfcMiFareTagTechnology.yml:41]: ```data``` parameter is not defined here

File: ios/Classes/TiNfcMiFareUltralightTagTechnology.m [hunk 4]
```diff
@@ -9,4 +9,98 @@
...
+  NSNumber *instructionCodeValue = [[args firstObject] valueForKey:@"instructionCode"];
+  uint8_t instructionCode = [instructionCodeValue unsignedCharValue];
+
+  NSNumber *p1ParameterValue = [[args firstObject] valueForKey:@"p1Parameter"];
+  uint8_t p1Parameter = [p1ParameterValue unsignedCharValue];
+
+  NSNumber *p2ParameterValue = [[args firstObject] valueForKey:@"p2Parameter"];
+  uint8_t p2Parameter = [p2ParameterValue unsignedCharValue];
+
+  NSNumber *expectedResponseLengthData = [[args objectAtIndex:0] valueForKey:@"expectedResponseLength"];
```
Comment (vikas-goyal-axway) [inline ios/Classes/TiNfcMiFareUltralightTagTechnology.m:82]:
```suggestion
  NSNumber *expectedResponseLengthData = [[args firstObject] valueForKey:@"expectedResponseLength"];
```

File: ios/Classes/TiNfcMiFareUltralightTagTechnology.m [hunk 5]
```diff
@@ -9,4 +9,98 @@
...
+                                        @"errorDomain" : [error domain],
+                                        @"response" : responseData
+                                      }];
+                               }
+                             }];
+}
+
+- (void)sendMiFareISO7816Command:(id)args
+{
+  NSArray *commandData = [[args objectAtIndex:0] valueForKey:@"data"];
```
Comment (vikas-goyal-axway) [inline ios/Classes/TiNfcMiFareUltralightTagTechnology.m:63]:
```suggestion
  NSArray *commandData = [[args firstObject] valueForKey:@"data"];
```

File: ios/Classes/TiNfcMiFareUltralightTagTechnology.m [hunk 6]
```diff
@@ -9,4 +9,98 @@
...
+{
+  TiBuffer *historicalBytes = [[TiBuffer alloc] _initWithPageContext:[self pageContext]];
+  NSMutableData *data = [NSMutableData dataWithData:[[self.tag asNFCMiFareTag] historicalBytes]];
+  [historicalBytes setData:data];
+  return historicalBytes;
+}
+
+- (void)sendMiFareCommand:(id)args
+{
+  NSArray *commandData = [[args objectAtIndex:0] valueForKey:@"data"];
```
Comment (vikas-goyal-axway) [inline ios/Classes/TiNfcMiFareUltralightTagTechnology.m:35]:
```suggestion
  NSArray *commandData = [[args firstObject] valueForKey:@"data"];
```

**Review by vikas-goyal-axway — CHANGES_REQUESTED** [review]

**Review by fasihul-axway — APPROVED** [review]

**Review by vikas-goyal-axway — APPROVED** [review]

**Pull request merged**